test/integration/acceptance/custom/hello_policy/hello_policy_test.go

```diff
@@ -0,0 +1,520 @@
+	"github.com/skupperproject/skupper/test/utils/skupper/cli"
+	"github.com/skupperproject/skupper/test/utils/skupper/cli/link"
+	"github.com/skupperproject/skupper/test/utils/skupper/cli/service"
+	"github.com/skupperproject/skupper/test/utils/skupper/cli/token"
+)
+
+// TestHelloPolicy is a test that runs the hello-world-example
+// scenario using just the "skupper" binary, which must be available
+// in the PATH.
+//
+// It is a copy of the test at test/integration/examples/custom/helloworld/,
+// adapted for Policy testing
+//
+// This test includes installation and removal of CRDs in-between other
+// steps.  For that reason, it runs only in one cluster, even if the
+// environment setup has two clusters.
+func testHelloPolicy(t *testing.T, pub1, pub2 *base.ClusterContext) {
+
+	// These test scenarios allow defining a set of skupper cli
+	// commands to be executed as a workflow, against specific
+	// clusters. Each execution is validated accordingly by its
+	// SkupperCommandTester implementation.
+	//
+	// The idea is to cover most of the main skupper commands
+	// as we run the hello-world-example so that all manipulation
+	// is performed just by the skupper binary, while each
+	// SkupperCommandTester implementation validates necessary
+	// output or resources in the cluster to certify the command
+	// was executed correctly.
+	initSteps := []cli.TestScenario{
+		skupperInitInteriorTestScenario(pub1, "", false),
+		skupperInitEdgeTestScenario(pub2, "", false),
+	}
+
+	connectSteps := cli.TestScenario{
+		Name: "connect-sites",
+		Tasks: []cli.SkupperTask{
+			{Ctx: pub1, Commands: []cli.SkupperCommandTester{
+				// skupper token create - verify token has been created
+				&token.CreateTester{
+					Name:     "public",
+					FileName: filepath.Join(testPath, "public-hello-world-1.token.yaml"),
+				},
+			}},
+			{Ctx: pub2, Commands: []cli.SkupperCommandTester{
+				// skupper link create - connect to public and verify connection created
+				&link.CreateTester{
+					TokenFile: filepath.Join(testPath, "public-hello-world-1.token.yaml"),
+					Name:      "public",
+					Cost:      1,
+				},
+			}},
+		},
+	}
+
+	//validateConnSteps Steps to confirm a link exists from the private namespace to the public one
+	validateConnSteps := cli.TestScenario{
+		Name: "validate-sites-connected",
+		Tasks: []cli.SkupperTask{
+			{Ctx: pub1, Commands: []cli.SkupperCommandTester{
+				// skupper status - verify sites are connected
+				&cli.StatusTester{
+					RouterMode:          "interior",
+					ConnectedSites:      1,
+					ConsoleEnabled:      true,
+					ConsoleAuthInternal: true,
+				},
+			}},
+			{Ctx: pub2, Commands: []cli.SkupperCommandTester{
+				// skupper status - verify sites are connected
+				&cli.StatusTester{
+					RouterMode:     "edge",
+					SiteName:       "private",
+					ConnectedSites: 1,
+				},
+				// skupper link status - testing all links
+				&link.StatusTester{
+					Name:   "public",
```

Comment:
The comment is incorrect. It is not testing all links. You can fix just this part and I believe
it would be nice to enhance it (in a separate PR later) to properly validate all links (not just
the auto-generated ones, but all).